clang/lib/CodeGen/CoverageMappingGen.cpp

```diff
@@ -1973,6 +1981,8 @@ struct CounterCoverageMappingBuilder
   void VisitBinLAnd(const BinaryOperator *E) {
     bool IsRootNode = MCDCBuilder.isIdle();
 
+    MCDCDebugCounter++;
```

Comment:
I didn't understand the strategy to assign it.

Reply:
Hi thanks again for taking the look!
My intuition was to make the ID different whenever a new decision region is going to be added by calling `createDecisionRegion()`. There might be something obvious that I ignored.
And after reading some of the other posts (I'm still learning most of them, please bear with that!), I generally agree with the approaches whose modifications are minimized within `llvm-cov`, instead of touching the front end.

Regards